src/representation_theory/group_cohomology_resolution.lean

```diff
@@ -60,145 +58,194 @@ over `k`.
+      exact fin.cases (by simp only [fin.cons_zero, fin.partial_prod_zero, mul_one]) (λ i,
+        by simp only [subsingleton.elim i 0, fin.cons_succ, pi.smul_apply, smul_eq_mul,
+          mul_right_inj, fin.partial_prod_succ, fin.cast_succ_zero,
+          fin.partial_prod_zero, one_mul]) x },
+    { ext,
+      dunfold Action_diagonal_succ,
+      simp only [iso.trans_inv, comp_hom, hn, diagonal_succ_inv_hom,
+        types_comp_apply, tensor_iso_inv, iso.refl_inv, tensor_hom, id_hom, mk_iso_inv_hom,
+        equiv.to_iso_inv, equiv.symm_symm, equiv.pi_fin_succ_above_equiv_apply,
+        fin.zero_succ_above, tensor_apply, types_id_apply,
+        left_regular_tensor_iso_inv_hom, tensor_rho, monoid_hom.one_apply, End.one_def,
+        left_regular_ρ_apply, pi.smul_apply, smul_eq_mul],
+      refine fin.cases _ _ x,
+      { simp only [fin.cons_zero, fin.partial_prod_zero, mul_one], },
+      { intro i,
+        simpa only [fin.cons_succ, pi.smul_apply, smul_eq_mul,
+          fin.partial_prod_succ', mul_assoc], }}}
 end
 
-lemma to_tensor_aux_of_mul_action (g : G) (x : Gⁿ⁺¹) :
-  to_tensor_aux k G n (of_mul_action k G Gⁿ⁺¹ g (single x 1)) =
-  tensor_product.map (of_mul_action k G G g) 1 (to_tensor_aux k G n (single x 1)) :=
-by simp [of_mul_action_def, to_tensor_aux_single, mul_assoc, inv_mul_cancel_left]
+end Action
+section Rep
+open Rep
 
-lemma of_tensor_aux_single (g : G) (m : k) (x : Gⁿ →₀ k) :
-  of_tensor_aux k G n ((single g m) ⊗ₜ x) =
-  finsupp.lift (Gⁿ⁺¹ →₀ k) k Gⁿ (λ f, single (g • partial_prod f) m) x :=
-by simp [of_tensor_aux, sum_single_index, smul_sum, mul_comm m]
+/-- An isomorphism of `k`-linear representations of `G` from `k[Gⁿ⁺¹]` to `k[G] ⊗ₖ k[Gⁿ]` (on
+which `G` acts by `ρ(g₁)(g₂ ⊗ x) = (g₁ * g₂) ⊗ x`) sending `(g₀, ..., gₙ)` to
+`g₀ ⊗ (g₀⁻¹g₁, g₁⁻¹g₂, ..., gₙ₋₁⁻¹gₙ)`. -/
+def diagonal_succ (n : ℕ) :
+  diagonal k G (n + 1) ≅ left_regular k G ⊗ trivial k G ((fin n → G) →₀ k) :=
+(linearization k G).1.1.map_iso (Action_diagonal_succ G n)
+  ≪≫ (as_iso ((linearization k G).μ (Action.left_regular G) _)).symm
+  ≪≫ tensor_iso (iso.refl _) (linearization_trivial_iso k G (fin n → G))
 
-lemma of_tensor_aux_comm_of_mul_action (g h : G) (x : Gⁿ) :
-  of_tensor_aux k G n (tensor_product.map (of_mul_action k G G g)
-  (1 : module.End k (Gⁿ →₀ k)) (single h (1 : k) ⊗ₜ single x (1 : k))) =
-  of_mul_action k G Gⁿ⁺¹ g (of_tensor_aux k G n (single h 1 ⊗ₜ single x 1)) :=
+variables {k G n}
+
+lemma diagonal_succ_hom_single (f : Gⁿ⁺¹) (a : k) :
+  (diagonal_succ k G n).hom.hom (single f a) =
+  single (f 0) 1 ⊗ₜ single (λ i, (f i)⁻¹ * f i.succ) a :=
 begin
-  simp [of_mul_action_def, of_tensor_aux_single, mul_smul],
+  dunfold diagonal_succ,
+  simp only [iso.trans_hom, iso.symm_hom, iso.refl_hom, tensor_iso_hom, Action.tensor_hom,
+    Action.comp_hom, Module.comp_def, linear_map.comp_apply, functor.map_iso_hom,
+    functor.map_comp],
+  rw linearization_map_hom, -- why does it time out when I `simp only` this?
```

Comment:
It doesn't time out for me, but it does take longer (the simp traces seem to indicate that it's spending a while failing to do some potential unification). Maybe add some explicit inputs to fix this if you want to get it into the previous line?